dist/rules/no-internal.js

```diff
@@ -342,12 +346,20 @@ module.exports = {
         if (!tsNode) return;
 
         const resolvedModule = typeChecker.getSymbolAtLocation(tsNode.moduleSpecifier);
-        if (resolvedModule && resolvedModule.exports) {
-          resolvedModule.exports.forEach((exportSymbol) => {
-            if (exportSymbol.valueDeclaration) {
-              checkWithParent(exportSymbol.valueDeclaration, node);
+        if (!resolvedModule || !resolvedModule.exports) return;
+
+        // Iterate over the named imports and default import
+        if (node.specifiers) {
+          for (const specifier of node.specifiers) {
+            if (specifier.type === "ImportSpecifier" || specifier.type === "ImportDefaultSpecifier" || specifier.type === "ImportNamespaceSpecifier") {
+              const importedName = specifier.local.name;
```

Comment:
Sorry, I found another loophole:
![image](https://github.com/user-attachments/assets/8bac962b-9046-41ee-aa49-3341740b3045)
Changing `local` to `imported` should fix that

Reply:
updated but made it a ternary operator because it is needed for this case where it is `import * as x from "y"`
![image](https://github.com/user-attachments/assets/e680d5e7-b9d6-4f51-92b5-7996f8aab3a1)

and your loophole is patched

Reply:
+ updated test to catch these cases